### PR TITLE
feat: use ECMAScript modules (ESM) instead of CommonJS

### PR DESCRIPTION
**Pull request by ssaarela**
BREAKING CHANGE

### CODE REVIEW

**Review by karlkeskyla — APPROVED** [review]
Neat! 🎉

File: packages/core/package.json [hunk 1]
```diff
@@ -3,6 +3,15 @@
...
+  "exports": {
+    ".": {
+      "import": {
+        "types": "./dist/index.d.ts",
+        "default": "./dist/index.js"
+      }
+    }
+  },
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
```
Comment (jlehtiniemi) [inline packages/core/package.json:16]: The CJS packaging definitions should probably be removed when building ESM-only package

File: packages/core/src/schema.spec.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
-import { SchemaValidator, DiscriminatorViolation } from './schema';
+import { describe, test, expect } from 'vitest'
+import { SchemaValidator, DiscriminatorViolation } from './schema.js';
 import { V } from './V';
-import { defaultViolations, TypeMismatch, ObjectValidator, HasValueViolation, Violation } from './validators';
+import { defaultViolations, TypeMismatch, ObjectValidator, HasValueViolation, Violation } from './validators.js';
 import { expectViolations, expectValid } from './testUtil.spec';
```
Comment (jlehtiniemi) [inline packages/core/src/schema.spec.ts:5]: Check the `.js` suffixes for the local imports

File: packages/luxon/package.json [hunk 3]
```diff
@@ -3,6 +3,15 @@
...
+  "exports": {
+    ".": {
+      "import": {
+        "types": "./dist/index.d.ts",
+        "default": "./dist/index.js"
+      }
+    }
+  },
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
```
Comment (jlehtiniemi) [inline packages/luxon/package.json:16]: Remove CJS packaging definitions

File: packages/moment/package.json [hunk 4]
```diff
@@ -3,6 +3,15 @@
...
+  "exports": {
+    ".": {
+      "import": {
+        "types": "./dist/index.d.ts",
+        "default": "./dist/index.js"
+      }
+    }
+  },
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
```
Comment (jlehtiniemi) [inline packages/moment/package.json:16]: Remove CJS packaging defs

File: packages/path-parser/package.json [hunk 5]
```diff
@@ -3,6 +3,15 @@
...
+  "exports": {
+    ".": {
+      "import": {
+        "types": "./dist/index.d.ts",
+        "default": "./dist/index.js"
+      }
+    }
+  },
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
```
Comment (jlehtiniemi) [inline packages/path-parser/package.json:16]: Remove CJS packaging defs

File: packages/path/package.json [hunk 6]
```diff
@@ -3,6 +3,15 @@
...
+  "exports": {
+    ".": {
+      "import": {
+        "types": "./dist/index.d.ts",
+        "default": "./dist/index.js"
+      }
+    }
+  },
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
```
Comment (jlehtiniemi) [inline packages/path/package.json:16]: Remove CJS packaging defs

**Review by jlehtiniemi — APPROVED** [review]

**Pull request merged**